src/viam/trajex/totg/trajectory.cpp

```diff
@@ -370,66 +373,87 @@ enum class integration_event : std::uint8_t {
+        // Compute max acceleration at switching velocity on both sides
+        const auto [_3, s_ddot_max_before] =
             compute_acceleration_bounds(q_prime_before, q_double_prime_before, s_dot_switching, opt.max_acceleration, opt.epsilon);
 
-        const auto [s_ddot_min_after, s_ddot_max_after] =
+        const auto [_4, s_ddot_max_after] =
             compute_acceleration_bounds(q_prime_after, q_double_prime_after, s_dot_switching, opt.max_acceleration, opt.epsilon);
 
-        // Check that we can pass through this point in forward direction
-        // For a discontinuous switching point with piecewise-constant curvature, we need:
-        // - Positive acceleration available on both sides (can reach from left, continue on right)
-        // - The switching velocity is at the acceleration limit curve on at least one side (touching the limit)
-        const bool can_reach_from_left = s_ddot_max_before > -opt.epsilon;
-        const bool can_continue_on_right = s_ddot_max_after > -opt.epsilon;
-        const bool touches_mvc = (std::abs(s_dot_switching - s_dot_max_acc_before) < opt.epsilon) ||
-                                 (std::abs(s_dot_switching - s_dot_max_acc_after) < opt.epsilon);
-
-        // TODO: Equation 38: Validate that this discontinuity is a trajectory SINK.
-        // A discontinuity is a valid switching point only if the maximum acceleration trajectory
-        // flows into it (is a sink), not away from it (source). We check this by comparing
-        // the trajectory slope (s_ddot_max/s_dot) with the limit curve slope (d/ds s_dot_max_acc).
-
-        if (can_reach_from_left && can_continue_on_right && touches_mvc) {
+        // Compute limit curve slopes using numerical approximation
+        const arc_length before_boundary = std::max(boundary - arc_length{opt.epsilon}, current_segment.start());
+        const auto q_prime_bb = current_segment.tangent(before_boundary);
+        const auto q_double_prime_bb = current_segment.curvature(before_boundary);
+        const auto [s_dot_bb, _5] =
+            compute_velocity_limits(q_prime_bb, q_double_prime_bb, opt.max_velocity, opt.max_acceleration, opt.epsilon);
+        const double slope_left = (s_dot_max_acc_before - s_dot_bb) / opt.epsilon;
+
+        const arc_length after_boundary = std::min(boundary + arc_length{opt.epsilon}, segment_after.end());
+        const auto q_prime_aa = segment_after.tangent(after_boundary);
+        const auto q_double_prime_aa = segment_after.curvature(after_boundary);
+        const auto [s_dot_aa, _6] =
+            compute_velocity_limits(q_prime_aa, q_double_prime_aa, opt.max_velocity, opt.max_acceleration, opt.epsilon);
+        const double slope_right = (s_dot_aa - s_dot_max_acc_after) / opt.epsilon;
+
+        // === Apply Equation 38 ===
+        bool is_switching_point = false;
+
+        // Case A: Positive step (limit increases)
+        if (s_dot_max_acc_before < s_dot_max_acc_after - opt.epsilon) {
```

Comment:
I know the generated code doesn't do this, but I strongly prefer not using epsilon in computations, and instead doing computations and then comparing the result to epsilon.